app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,11 @@
+class MerchantsController < ApplicationController
+  @@facade = HeaderFacade.new
```

Comment:
This is a very interesting approach and I love that you experimented with this. I don't often see class variables in ruby, so this was fun to see. I think I would've approached how to access the `HeaderFacade` a little differently, but I like your idea of using one facade that can be used with all endpoints. I think you might find that if you were to start building out more features and views that the name `HeaderFacade` is a little vague, so something like `MerchantFacade` might still be a better name. Alternatively, something like `RailsEngineFacade` is still broadly name but also directly tied to the API/service. 

Anyways, back to how we're accessing the facade. I think instead of using a class var I'd instantiate as a private method and call that from within the `index` and `show` actions. I think that's just a more typical way of how you'd see a rails project do this, but your way is not wrong by any means. I will add the code for how I would do this on the appropriate lines.